account_payment_partner/models/account_payment_mode.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountPaymentMode(models.Model):
+    _inherit = 'account.payment.mode'
+
+    show_bank_account = fields.Selection(
+        selection=[
+            ('full', 'Full'),
+            ('first', 'First n chars'),
+            ('last', 'Last n chars'),
+            ('no', 'No'),
+        ],
+        string='Show bank account',
+        default='full',
+        help="Show in invoices partial or full bank account number")
```

Comment:
@alexis-via I don't think this is too specific: this allows to print in the invoice to which account number the payment (via SEPA direct debit) is going to be charged so the customer knows that information, but there are some customers that don't want that everyone looking at the invoice sees the full account number (and even GDPR can say a word about that), so that's the reason behind the several options.

Having in another module is something oversized as the module will be here in bank-payment also, and there's not too much code about this. If you don't want to use it, just leave the default option. I promise we will maintain this code in future versions :wink:

Reply:
Ah, ok, I understand the reason for this feature. I think it should be explained more clearly!

Reply:
I think we have a problem with the field 'partner_bank_id 'of account.invoice for bank accounts. In the "account" module, the help message of this field says: "Bank Account Number to which the invoice will be paid. A Company bank account if this is a Customer Invoice or Vendor Refund, otherwise a Partner bank account number."
I think we cannot change the meaning of this field i.e. for a customer invoice paid by sepa direct debit, we can't store the Partner bank account number in this field. And it seems to me that it is what you are doing. So I think we need another approach: we should inherit the invoice report both in account_payment_partner and also in account_banking_mandate (in account_banking_mandate, we would display mandate_id.partner_bank_id).

BTW, that would be a good topic for during the next OCA code sprint in Sevilla (I would be ready to work on this)... but we urgently need a date for this code sprint, because my customers are already starting to book my work days in June...

Reply:
Yeah, I'm pushing for having the date ASAP, but we rely on the organizer there in Sevilla for that. I'll try to push even further.